### PR TITLE
Create a file for the base classes for Formatters

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
Refactor prior to #15

### CODE REVIEW

**Review by DanielNoord — COMMENTED** [review]
After all the work you did I need to take another good look at what is private and isn't, but I think it would make sense to expose these two classes in the public API.

File: pydocstringformatter/formatting/base.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import abc
+import tokenize
+
+
+class Formatter:
+    """Base class for docstring formatter"""
```
Comment (DanielNoord) [inline pydocstringformatter/formatting/base.py:6]:
```suggestion
    """Base class for docstring formatter."""
```

File: pydocstringformatter/formatting/base.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+    @abc.abstractmethod
+    def treat_token(self, tokeninfo: tokenize.TokenInfo) -> tokenize.TokenInfo:
+        """Return a modified token"""
```
Comment (DanielNoord) [inline pydocstringformatter/formatting/base.py:10]:
```suggestion
        """Return a modified token."""
```

File: pydocstringformatter/formatting/base.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+
+class StringFormatter(Formatter):
+    """Base class for formatter that only modifies the string content"""
+
+    @abc.abstractmethod
+    def _treat_string(self, tokeninfo: tokenize.TokenInfo) -> str:
+        """Return a modified string"""
```
Comment (DanielNoord) [inline pydocstringformatter/formatting/base.py:18]:
```suggestion
        """Return a modified string."""
```

File: pydocstringformatter/formatting/base.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+class Formatter:
+    """Base class for docstring formatter"""
+
+    @abc.abstractmethod
+    def treat_token(self, tokeninfo: tokenize.TokenInfo) -> tokenize.TokenInfo:
+        """Return a modified token"""
+
+
+class StringFormatter(Formatter):
+    """Base class for formatter that only modifies the string content"""
```
Comment (DanielNoord) [inline pydocstringformatter/formatting/base.py:14]:
```suggestion
    """Base class for formatter that only modifies the string content."""
```

**Review by DanielNoord — APPROVED** [review]

File: pydocstringformatter/formatting/base.py (hunk 2 above)
Comment (Pierre-Sassoulas) [reply pydocstringformatter/formatting/base.py:10]: Huhu, you can't wait for this new DotFormatter huh 😄

Comment (DanielNoord) [reply pydocstringformatter/formatting/base.py]: Well, I'm actually a bit hesitant. I would like to do something like "CheckIfSentenceFormatter" first, as adding a dot to non-sentences is weird and might shy users away from using the tool.

Perhaps I can leverage the work you're doing in #15 to make it an optional formatter to begin with 😄

Comment (Pierre-Sassoulas) [reply pydocstringformatter/formatting/base.py]: Yes it's non trivial, I think a first version could add a line feed after existing dot and add a dot to string not ending with ``:``.

**Pull request merged**